internal/pkg/service/appsproxy/dataapps/appconfig/middleware.go

```diff
@@ -0,0 +1,88 @@
+package appconfig
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	"go.opentelemetry.io/otel/attribute"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/service/appsproxy/dataapps/api"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/ctxattr"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/httpserver/middleware"
+)
+
+type ctxKey string
+
+const (
+	appConfigCtxKey = ctxKey("app-config")
+)
+
+type AppConfigResult struct {
+	AppID     api.AppID
+	AppConfig api.AppConfig
+	Modified  bool
+	Err       error
+}
+
+func AppConfigFromContext(ctx context.Context) AppConfigResult {
+	appConfig := ctx.Value(appConfigCtxKey)
+	if appConfig == nil {
+		return AppConfigResult{}
+	}
+	return appConfig.(AppConfigResult)
```

Comment:
```suggestion
	if appConfig, ok := ctx.Value(appConfigCtxKey).(AppConfigResult); ok {
		return appConfig
	}
	return AppConfigResult{}
```
This will help you avoid unnecessary panics in your code.